mindee/input/local_response.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+import hmac
+import io
+import json
+from pathlib import Path
+from typing import Any, BinaryIO, Dict, Union
+
+from mindee.error import MindeeError
+
+
+class LocalResponse:
+    """Local response loaded from a file."""
+
+    _file: BinaryIO
+    """File object of the local response."""
+
+    def __init__(self, input_file: Union[BinaryIO, str, Path, bytes]):
+        if isinstance(input_file, BinaryIO):
+            self._file = input_file
+            self._file.seek(0)
+        elif isinstance(input_file, (str, Path)):
+            with open(input_file, "r", encoding="utf-8") as file:
+                self._file = io.BytesIO(
+                    file.read().replace("\r", "").replace("\n", "").encode()
+                )
+        elif isinstance(input_file, bytes):
+            self._file = io.BytesIO(input_file)
+        else:
+            raise MindeeError("Incompatible type for input.")
+
+    @property
+    def as_dict(self) -> Dict[str, Any]:
+        """
+        Returns the dictionary representation of the file.
+
+        :return: A json-like dictionary.
+        """
+        try:
+            self._file.seek(0)
+            out_json = json.loads(self._file.read())
+        except json.decoder.JSONDecodeError as exc:
+            raise MindeeError("File is not a valid dictionary.") from exc
+        return out_json
+
+    @staticmethod
+    def _process_secret_key(
+        secret_key: Union[str, bytes, bytearray]
+    ) -> Union[bytes, bytearray]:
+        """
+        Processes the secret key as a byte array.
+
+        :param secret_key: Secret key, either a string or a byte/byte array.
+        :return: a byte/byte array secret key.
+        """
+        if isinstance(secret_key, (bytes, bytearray)):
+            return secret_key
+        return secret_key.encode("utf-8")
+
+    def get_hmac_signature(self, secret_key: Union[str, bytes, bytearray]):
```

Comment:
Not sure why anyone would send the secret key as anything other than a string, but OK, why not.

Reply:
`hmac.new()` takes in a sequence of bytes, I was thinking of people who would try and use it like this

Reply:
:+1: